app/src/main/kotlin/com/adesso/movee/scene/tvshow/TvShowViewModel.kt

```diff
@@ -48,12 +48,8 @@ class TvShowViewModel @Inject constructor(
     }
 
     private fun fetchTopRatedTvShows() {
-        bgScope.launch {
-            val topRatedTvShowsResult = fetchTopRatedTvShowsUseCase.run(UseCase.None)
-
-            onUIThread {
-                topRatedTvShowsResult.either(::handleFailure, ::postTopRatedTvShows)
-            }
+        fetchTopRatedTvShowsUseCase.invoke(viewModelScope, UseCase.None) {
+            it.either(::handleFailure, ::postTopRatedTvShows)
```

Comment:
postTopRatedTvShows() uses LiveData's setValue() function which must be called from main thread. Either we need to use postValue(), or switch context when calling postTopRatedTvShows()